tests/unit/validation/test_validation_functions.py

```diff
@@ -113,7 +121,7 @@ def test_validate_unique_ids_across_components():
     assert len(unique_id_errors[0].ids) == 4
 
 
-def test_validate_ids_exist():
+def test_validate_ids():
```

Comment:
This reminded me: We don't test the `validate_batch_data` function with update data without the Ids, do we?

Reply:
We don't. Not sure if the current validation logic can handle that deserialization. Could be a nice follow-up.

Reply:
The current logic doesn't support it indeed. So that is something additional to add. However, we should probably throw a not implemented error at least for now, right? Or else it would just plainly fail on the users side.

Reply:
The validation batch tests is internal test for dev, not for users. Users' code is totally possible to be configured to deserialize to columnar from json and run pf (with or without ids). Adding such a test in validation batch test is todo.

Reply:
As discussed, conversation resolved.